sdk/compiler/daml-lf-proto-util/src/DA/Daml/LF/Proto3/DerivingData.hs

```diff
@@ -0,0 +1,146 @@
+deriving instance Data P.Expr_TyAbs
+deriving instance Data P.Expr_Nil
+deriving instance Data P.Expr_Cons
+deriving instance Data P.Expr_OptionalNone
+deriving instance Data P.Expr_OptionalSome
+deriving instance Data P.Expr_ToAny
+deriving instance Data P.Expr_FromAny
+deriving instance Data P.Expr_ToAnyException
+deriving instance Data P.Expr_FromAnyException
+deriving instance Data P.Expr_Throw
+deriving instance Data P.Expr_ToInterface
+deriving instance Data P.Expr_FromInterface
+deriving instance Data P.Expr_CallInterface
+deriving instance Data P.Expr_ViewInterface
+deriving instance Data P.Expr_SignatoryInterface
+deriving instance Data P.Expr_ObserverInterface
+deriving instance Data P.Expr_UnsafeFromInterface
+deriving instance Data P.Expr_ToRequiredInterface
+deriving instance Data P.Expr_FromRequiredInterface
+deriving instance Data P.Expr_UnsafeFromRequiredInterface
+deriving instance Data P.Expr_InterfaceTemplateTypeRep
+deriving instance Data P.Expr_ChoiceController
+deriving instance Data P.Expr_ChoiceObserver
+deriving instance Data P.Expr_Experimental
+deriving instance Data P.ExprSum
+deriving instance Data P.CaseAlt
+deriving instance Data P.CaseAlt_Variant
+deriving instance Data P.CaseAlt_Enum
+deriving instance Data P.CaseAlt_Cons
+deriving instance Data P.CaseAlt_OptionalSome
+deriving instance Data P.CaseAltSum
+deriving instance Data P.Case
+deriving instance Data P.Block
+deriving instance Data P.Pure
+deriving instance Data P.Update
+deriving instance Data P.Update_Create
+deriving instance Data P.Update_CreateInterface
+deriving instance Data P.Update_Exercise
+deriving instance Data P.Update_ExerciseInterface
+deriving instance Data P.Update_ExerciseByKey
+deriving instance Data P.Update_Fetch
+deriving instance Data P.Update_FetchInterface
+deriving instance Data P.Update_EmbedExpr
+deriving instance Data P.Update_RetrieveByKey
+deriving instance Data P.Update_TryCatch
+deriving instance Data P.UpdateSum
+deriving instance Data P.TemplateChoice
+deriving instance Data P.InterfaceInstanceBody
+deriving instance Data P.InterfaceInstanceBody_InterfaceInstanceMethod
+deriving instance Data P.DefTemplate
+deriving instance Data P.DefTemplate_DefKey
+deriving instance Data P.DefTemplate_Implements
+deriving instance Data P.InterfaceMethod
+deriving instance Data P.DefInterface
+deriving instance Data P.DefException
+deriving instance Data P.DefDataType
+deriving instance Data P.DefDataType_Fields
+deriving instance Data P.DefDataType_EnumConstructors
+deriving instance Data P.DefDataTypeDataCons
+deriving instance Data P.DefTypeSyn
+deriving instance Data P.DefValue
+deriving instance Data P.DefValue_NameWithType
+deriving instance Data P.FeatureFlags
+deriving instance Data P.Module
+deriving instance Data P.InternedDottedName
+deriving instance Data P.UpgradedPackageId
+deriving instance Data P.Package
+deriving instance Data P.PackageMetadata
+
+instance Data a => Data (P.Enumerated a) where
```

Comment:
A comment explaining what's going on here would be useful for those who (like me) aren't as familiar with Proto3.Suite internals.